#**Parte 2 — Exploratory Data Analysis (EDA)**



##1. Carga y revisión inicial

In [1]:
import pandas as pd
import numpy as np

import nltk
from nltk.corpus import stopwords

import re
import string

In [2]:
# Ruta al archivo (ajusta el nombre del archivo)
file_path = "/content/Amazon_Consumer_Review.csv"

# Cargar el dataset en Pandas
df = pd.read_csv(file_path)

# Ver las primeras filas
df.head()

,id,dateAdded,dateUpdated,name,asins,brand,categories,primaryCategories,imageURLs,keys,...,reviews.didPurchase,reviews.doRecommend,reviews.id,reviews.numHelpful,reviews.rating,reviews.sourceURLs,reviews.text,reviews.title,reviews.username,sourceURLs
0,AVpgNzjwLJeJML43Kpxn,2015-10-30T08:59:32Z,2019-04-25T09:08:16Z,AmazonBasics AAA Performance Alkaline Batterie...,"B00QWO9P0O,B00LH3DMUO",Amazonbasics,"AA,AAA,Health,Electronics,Health & Household,C...",Health & Beauty,https://images-na.ssl-images-amazon.com/images...,"amazonbasics/hl002619,amazonbasicsaaaperforman...",...,NaN,NaN,NaN,NaN,3,https://www.amazon.com/product-reviews/B00QWO9...,I order 3 of them and one of the item is bad q...,... 3 of them and one of the item is bad quali...,Byger yang,"https://www.barcodable.com/upc/841710106442,ht..."
1,AVpgNzjwLJeJML43Kpxn,2015-10-30T08:59:32Z,2019-04-25T09:08:16Z,AmazonBasics AAA Performance Alkaline Batterie...,"B00QWO9P0O,B00LH3DMUO",Amazonbasics,"AA,AAA,Health,Electronics,Health & Household,C...",Health & Beauty,https://images-na.ssl-images-amazon.com/images...,"amazonbasics/hl002619,amazonbasicsaaaperforman...",...,NaN,NaN,NaN,NaN,4,https://www.amazon.com/product-reviews/B00QWO9...,Bulk is always the less expensive way to go fo...,... always the less expensive way to go for pr...,ByMG,"https://www.barcodable.com/upc/841710106442,ht..."
2,AVpgNzjwLJeJML43Kpxn,2015-10-30T08:59:32Z,2019-04-25T09:08:16Z,AmazonBasics AAA Performance Alkaline Batterie...,"B00QWO9P0O,B00LH3DMUO",Amazonbasics,"AA,AAA,Health,Electronics,Health & Household,C...",Health & Beauty,https://images-na.ssl-images-amazon.com/images...,"amazonbasics/hl002619,amazonbasicsaaaperforman...",...,NaN,NaN,NaN,NaN,5,https://www.amazon.com/product-reviews/B00QWO9...,Well they are not Duracell but for the price i...,... are not Duracell but for the price i am ha...,BySharon Lambert,"https://www.barcodable.com/upc/841710106442,ht..."
3,AVpgNzjwLJeJML43Kpxn,2015-10-30T08:59:32Z,2019-04-25T09:08:16Z,AmazonBasics AAA Performance Alkaline Batterie...,"B00QWO9P0O,B00LH3DMUO",Amazonbasics,"AA,AAA,Health,Electronics,Health & Household,C...",Health & Beauty,https://images-na.ssl-images-amazon.com/images...,"amazonbasics/hl002619,amazonbasicsaaaperforman...",...,NaN,NaN,NaN,NaN,5,https://www.amazon.com/product-reviews/B00QWO9...,Seem to work as well as name brand batteries a...,... as well as name brand batteries at a much ...,Bymark sexson,"https://www.barcodable.com/upc/841710106442,ht..."
4,AVpgNzjwLJeJML43Kpxn,2015-10-30T08:59:32Z,2019-04-25T09:08:16Z,AmazonBasics AAA Performance Alkaline Batterie...,"B00QWO9P0O,B00LH3DMUO",Amazonbasics,"AA,AAA,Health,Electronics,Health & Household,C...",Health & Beauty,https://images-na.ssl-images-amazon.com/images...,"amazonbasics/hl002619,amazonbasicsaaaperforman...",...,NaN,NaN,NaN,NaN,5,https://www.amazon.com/product-reviews/B00QWO9...,These batteries are very long lasting the pric...,... batteries are very long lasting the price ...,Bylinda,"https://www.barcodable.com/upc/841710106442,ht..."


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28332 entries, 0 to 28331
Data columns (total 24 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   28332 non-null  object 
 1   dateAdded            28332 non-null  object 
 2   dateUpdated          28332 non-null  object 
 3   name                 28332 non-null  object 
 4   asins                28332 non-null  object 
 5   brand                28332 non-null  object 
 6   categories           28332 non-null  object 
 7   primaryCategories    28332 non-null  object 
 8   imageURLs            28332 non-null  object 
 9   keys                 28332 non-null  object 
 10  manufacturer         28332 non-null  object 
 11  manufacturerNumber   28332 non-null  object 
 12  reviews.date         28332 non-null  object 
 13  reviews.dateSeen     28332 non-null  object 
 14  reviews.didPurchase  9 non-null      object 
 15  reviews.doRecommend  16086 non-null 

In [4]:
df.describe()

,reviews.id,reviews.numHelpful,reviews.rating
count,4.100000e+01,16115.000000,28332.000000
mean,1.840066e+08,0.529321,4.514048
std,2.337036e+07,9.345017,0.934957
min,1.116244e+08,0.000000,1.000000
25%,1.843344e+08,0.000000,4.000000
50%,1.885078e+08,0.000000,5.000000
75%,1.988160e+08,0.000000,5.000000
max,2.085304e+08,621.000000,5.000000


In [5]:
df.dtypes

,0
id,object
dateAdded,object
dateUpdated,object
name,object
asins,object
brand,object
categories,object
primaryCategories,object
imageURLs,object
keys,object


##Información general del dataset

El dataset anexado corresponde a una base de datos de reseñas de consumidores sobre productos Amazon. Cada fila representa una opinión escrita por un cliente sobre un producto específico, junto con información asociada como calificación, fecha, producto, categoría, marca, recomendación y votos de utilidad.

Este dataset es adecuado para el proyecto porque contiene texto de reseñas, calificaciones numéricas y variables complementarias que permiten desarrollar un modelo de análisis de sentimiento con Machine Learning y NLP.

| Elemento                                   | Descripción                                                            |
| ------------------------------------------ | ---------------------------------------------------------------------- |
| **Nombre del archivo**                     | `Amazon_Consumer_Review.csv`                                           |
| **Formato**                                | CSV                                                                    |
| **Cantidad de filas**                      | 28,332 registros                                                       |
| **Cantidad de columnas**                   | 24 columnas                                                            |
| **Unidad de análisis**                     | Cada fila representa una reseña de un cliente sobre un producto Amazon |
| **Periodo de reseñas**                     | Aproximadamente de 2009 a 2019                                         |
| **Tipo de problema compatible**            | NLP y clasificación multiclase                                         |
| **Variable principal de texto**            | `reviews.text`                                                         |
| **Variable base para construir el target** | `reviews.rating`                                                       |



###¿Qué contiene el dataset?

El dataset contiene información relacionada con:

* Datos del producto
* Nombre del producto.
* Marca.
* Categoría.
* ASIN o identificador del producto.
Fabricante.
* URLs de imágenes y páginas fuente.
* Datos de la reseña
* Texto completo de la reseña.
* Título de la reseña.
* Calificación dada por el cliente.
* Fecha de la reseña.
* Nombre de usuario.
* Cantidad de votos de utilidad.
* Datos de recomendación
* Si el cliente recomienda o no el producto.
* Si el cliente compró el producto, aunque esta variable tiene muchos datos faltantes.

###**Variables mas importantes para el proyecto**

| Variable              | Descripción                                         | Uso en el proyecto                                                 |
| --------------------- | --------------------------------------------------- | ------------------------------------------------------------------ |
| `reviews.text`        | Texto completo de la reseña escrita por el cliente. | Variable principal para aplicar NLP.                               |
| `reviews.title`       | Título de la reseña.                                | Puede complementar el texto principal.                             |
| `reviews.rating`      | Calificación del producto de 1 a 5 estrellas.       | Se usará para construir el target de sentimiento.                  |
| `reviews.doRecommend` | Indica si el cliente recomienda el producto.        | Variable de apoyo para validar sentimiento.                        |
| `reviews.numHelpful`  | Número de votos de utilidad de la reseña.           | Puede analizarse para ver si las reseñas negativas son más útiles. |
| `reviews.date`        | Fecha de la reseña.                                 | Permite analizar cambios de sentimiento en el tiempo.              |
| `name`                | Nombre del producto reseñado.                       | Permite analizar sentimiento por producto.                         |
| `primaryCategories`   | Categoría principal del producto.                   | Permite analizar sentimiento por categoría.                        |
